torch/csrc/jit/passes/quantization.cpp

```diff
@@ -443,14 +443,22 @@ Node* insertQuantDeQuantCall(
 
   Node* quant = g->create(at::Symbol::aten(quantize_func), inputs);
   quant->output()->setDebugName(v->debugName() + ".quant");
-
-  Node* dequant = g->create(at::Symbol::aten("dequantize"), {quant->output()});
-  dequant->output()->setDebugName(v->debugName() + ".dequant");
-
   g->insertNode(quant);
-  g->insertNode(dequant);
 
-  return dequant;
+  std::vector<Node*> use_of_node_to_be_removed;
+  for (const auto& use : v->uses()) {
```

Comment:
I'm not quite sure what's going on here: what are we trying to do in this and the next loop?

Reply:
It just collects all the nodes using "v" and then replaces the input "v" with dequant output.

Reply:
Actually, the initial implementation just use one loop with the replacefirstusewith, which looks not that straightforward.

Reply:
So, this entire loop is to just put `quant->output()` to `use_nodes`? Maybe I'm missing something, but how can we have more than one element in `use_nodes`?

Reply:
Why didn't we do it like in the original code?
```
    v->replaceAllUsesWith(dequant->output());	
    Node* q = dequant->input(0)->node();	
    // replaceAllUsesWith rewrote all uses of V, but we want to keep one: the one	
    // used in quant node. Restore it here:	
    q->replaceInputWith(dequant->output(), v);
```

Reply:
You need to insert the dequant one by one, right?

Reply:
That's what I want to understand. Currently it looks like originally we have:
```
     [X]
      |
     (v)
    / | \
   /  |  \
 [A] [B] [C]
```

And we want to transform it to:
```
     [X]
      |
     (v)
      |
     [Q]
      |
    (vq )
      |
    [DQ ]
    / | \
   /  |  \
 [A] [B] [C]
```
If that's the case, then the original code works just fine. If it's something else, then I want to understand it and it should be clearly explained in comments.

It seems to me that the current code does exactly the same - if it's not, please explain what's different, if it is the same, please revert this part since the original code is much simpler.

Reply:
@ZolotukhinM we need multiple dequantize nodes for pattern matching

Reply:
Then this should be in the comments and in the commit message. What you said makes sense but it's literally impossible for the reader to recover that info - this kind of decisions *must* be reflected somewhere.

It's too late to change commit message now, so let's at least add the comments here.